src/components/blocks/software/Pre.tsx

```diff
@@ -1,26 +1,38 @@
-import React, { ReactElement, useContext } from 'react';
+import React, { ReactElement, useContext, useState } from 'react';
 import cn from 'classnames';
 import Icon from '@ably/ui/core/Icon';
 import Html from '../Html';
 import { PageLanguageContext } from 'src/contexts';
 import languageLabels from 'src/maps/language';
 import LocalLanguageAlternatives from '../wrappers/LocalLanguageAlternatives';
-import { DEFAULT_LANGUAGE } from '../../../../data/createPages/constants';
+import { DEFAULT_LANGUAGE, DEFAULT_PREFERRED_INTERFACE } from '../../../../data/createPages/constants';
 import { HtmlComponentProps, ValidReactElement } from '../../html-component-props';
 import HtmlDataTypes from '../../../../data/types/html';
 import { isString, every, reduce } from 'lodash/fp';
 import { MultilineCodeContent } from './Code/MultilineCodeContent';
-import { isArray } from 'lodash';
+import { isArray, isEmpty } from 'lodash';
 import { getFilteredLanguages } from 'src/components/common';
 
 type PreProps = HtmlComponentProps<'pre'> & {
   language: string;
   languages?: string[];
   altData?: Record<string, string | HtmlComponentProps<ValidReactElement>[] | null>;
+  isSDKInterface?: boolean;
+  realtimeAltData?: Record<string, string | HtmlComponentProps<ValidReactElement>[] | null>;
+  restAltData?: Record<string, string | HtmlComponentProps<ValidReactElement>[] | null>;
 };
 
-const Pre = ({ data, languages, altData, attribs }: PreProps): ReactElement => {
+const Pre = ({
+  data,
+  languages,
+  altData,
+  isSDKInterface = false,
+  realtimeAltData,
+  restAltData,
+  attribs,
+}: PreProps): ReactElement => {
   const pageLanguage = useContext(PageLanguageContext);
+
   const hasCode =
     languages?.some((lang) => getFilteredLanguages(lang) === pageLanguage) || pageLanguage === DEFAULT_LANGUAGE;
```

Comment:
Check somewhere around here for what the overall languages options are for a block of code (so you can see if you have `javascript` vs. `realtime_javascript` for example.